packages/ckeditor5-list/src/tododocumentlist/checktododocumentlistcommand.ts

```diff
@@ -0,0 +1,108 @@
+/**
+ * @license Copyright (c) 2003-2023, CKSource Holding sp. z o.o. All rights reserved.
+ * For licensing, see LICENSE.md or https://ckeditor.com/legal/ckeditor-oss-license
+ */
+
+/**
+ * @module list/tododocumentlist/checktododocumentlistcommand
+ */
+
+import { Command, type Editor } from 'ckeditor5/src/core';
+import type { Element } from 'ckeditor5/src/engine';
+import { getAllListItemBlocks } from '../documentlist/utils/model';
+
+const attributeKey = 'todoListChecked';
+
+/**
+ * The check to-do command.
+ *
+ * TODO
+ *
+ * The command is registered by the {@link module:list/todolist/todolistediting~TodoListEditing} as
+ * the `checkTodoList` editor command and it is also available via aliased `todoListCheck` name.
+ */
+export default class CheckTodoDocumentListCommand extends Command {
+	/**
+	 * A list of to-do list items selected by the {@link module:engine/model/selection~Selection}.
+	 *
+	 * @observable
+	 * @readonly
+	 */
+	declare public value: boolean;
+
+	/**
+	 * @inheritDoc
+	 */
+	constructor( editor: Editor ) {
+		super( editor );
+
+		// Refresh command before executing to be sure all values are up to date.
+		// It is needed when selection has changed before command execution, in the same change block.
+		this.on( 'execute', () => {
+			this.refresh();
+		}, { priority: 'highest' } );
+	}
+
+	/**
+	 * Updates the command's {@link #value} and {@link #isEnabled} properties based on the current selection.
+	 */
+	public override refresh(): void {
+		const selectedElements = this._getSelectedItems();
+
+		this.value = this._getValue( selectedElements );
+		this.isEnabled = !!selectedElements.length;
+	}
+
+	/**
+	 * Executes the command.
+	 *
+	 * @param options.forceValue If set, it will force the command behavior. If `true`, the command will apply
+	 * the attribute. Otherwise, the command will remove the attribute. If not set, the command will look for its current
+	 * value to decide what it should do.
+	 */
+	public override execute( options: { forceValue?: boolean } = {} ): void {
+		this.editor.model.change( writer => {
+			const selectedElements = this._getSelectedItems();
+			const value = ( options.forceValue === undefined ) ? !this._getValue( selectedElements ) : options.forceValue;
+
+			for ( const element of selectedElements ) {
+				if ( value ) {
+					writer.setAttribute( attributeKey, true, element );
+				} else {
+					writer.removeAttribute( attributeKey, element );
+				}
+			}
+		} );
+	}
+
+	/**
+	 * TODO
+	 */
+	private _getValue( selectedElements: Array<Element> ): boolean {
+		return selectedElements.every( element => !!element.getAttribute( attributeKey ) );
```

Comment:
```suggestion
		return selectedElements.every( element => element.getAttribute( attributeKey ) );
```
I believe, the expected return type of the predicate is `unknown`.